packages/modular-scripts/src/__tests__/addPackage.test.ts

```diff
@@ -2,91 +2,84 @@ import execa from 'execa';
 import rimraf from 'rimraf';
 import path from 'path';
 import fs from 'fs-extra';
+import tree from 'tree-view-for-tests';
 
 import getModularRoot from '../utils/getModularRoot';
 
 import type { CoreProperties } from '@schemastore/package';
+import { createModularTestContext, mockInstallTemplates } from '../test/utils';
 
 const modularRoot = getModularRoot();
 
-// These tests must be executed sequentially with `--runInBand`.
+// Setup temporary test context
+const tempModularRepo = createModularTestContext();
+console.log(`Temp Modular Repo: ${tempModularRepo}`);
+if (fs.existsSync(tempModularRepo)) {
+  console.log('Exists');
+}
+const packagesPath = path.join(tempModularRepo, 'packages');
+
+const templatesPath = path.join(modularRoot, '__fixtures__', 'templates');
+mockInstallTemplates(templatesPath, tempModularRepo);
 
-const packagesPath = path.join(getModularRoot(), 'packages');
+/**
+ * Reset 'packages' workspace in temporary Modular Repo
+ */
+function cleanup() {
+  rimraf.sync(packagesPath);
+  fs.mkdirSync(packagesPath);
+}
 
 function modular(str: string, opts: Record<string, unknown> = {}) {
-  return execa('yarnpkg', ['modular', ...str.split(' ')], {
-    cwd: modularRoot,
+  return execa('yarnpkg', ['modular', ...str.split(' '), '--verbose'], {
+    cwd: tempModularRepo,
     cleanup: true,
+    stdio: 'inherit',
     ...opts,
   });
 }
 
-function cleanup() {
-  rimraf.sync(path.join(packagesPath, 'sample-app'));
-  rimraf.sync(path.join(packagesPath, 'nested', 'scoped'));
-
-  // run yarn so yarn.lock gets reset
-  return execa.sync('yarnpkg', ['--silent'], {
-    cwd: modularRoot,
-  });
-}
-
-beforeAll(cleanup);
-afterAll(cleanup);
-
 describe('When setting a base directory for an app', () => {
   it('fails if trying to add an app outside the "workspaces" directories', async () => {
     await expect(
       modular(
         'add @scoped/will-not-create-app --path some/other/basepath --unstable-type app',
-        {
-          stdio: 'inherit',
-        },
       ),
     ).rejects.toThrow();
   });
 });
 
 describe('When working with a scoped app', () => {
   beforeAll(async () => {
-    await modular('add @scoped/sample-app --unstable-type app', {
-      stdio: 'inherit',
-    });
+    await modular('add @scoped/sample-app --unstable-type app');
```

Comment:
As this is in a `beforeAll` rather than `beforeEach` the tests will all use the same temp directory, meaning they could depend on the state left by previous tests which makes it less clear what is being tested.

As the simplest option, `createModularTestContext` in a `beforeEach` to set up the test directory for each test. If you find that the tests are taking too long you could create a context, add the app to it then clone that directory in a `beforeEach`

Reply:
That was my intention for performance reasons, but I agree beforeEach would provide a more 'solid'/isolated test context - I'll check the performance impact and explore your cloning suggestion, thanks

Reply:
Performance difference setting up the directory for each test was negligible so went ahead and made the change 
The test context is generated in the BeforeAll for each test as the tests were written in such a way that expects the results of previous assertion to still be there, see:

```
describe('When working with a scoped app', () => {
  beforeAll(async () => {
    createTempModularRepoWithTemplate(appTemplatePath);
    await modular('add @scoped/sample-app --unstable-type app');
  });

  it('creates the app in the expected directory, with the expected name', async () => {
    const manifest = (await fs.readJSON(
      path.join(tempPackagesPath, 'sample-app', 'package.json'),
    )) as CoreProperties;
    expect(manifest.name).toEqual('@scoped/sample-app');
  });

  it('fails if trying to add another app with the same name', async () => {
    await expect(
      modular('add @scoped/sample-app --unstable-type app'),
    ).rejects.toThrow();
  });
```